tests/Functional/Schema/MySQL/ComparatorTest.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace Doctrine\DBAL\Tests\Functional\Schema\MySQL;
+
+use Doctrine\DBAL\Exception;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Platforms\MySQLPlatform;
+use Doctrine\DBAL\Schema\AbstractSchemaManager;
+use Doctrine\DBAL\Schema\Column;
+use Doctrine\DBAL\Schema\Comparator;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Tests\Functional\Schema\ComparatorTestUtils;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+use Doctrine\DBAL\Types\Types;
+
+final class ComparatorTest extends FunctionalTestCase
+{
+    /** @var AbstractPlatform */
+    private $platform;
+
+    /** @var AbstractSchemaManager */
```

Comment:
```suggestion
    /** @var AbstractSchemaManager<MySQLPlatform> */
```

Reply:
This will require `Connection::createSchemaManager()` and `Driver::createSchemaManager()` to be parametrized which will lead to the same issues as in https://github.com/doctrine/dbal/pull/4746#discussion_r695024202.